include/boost/json/basic_parser.hpp

```diff
@@ -272,105 +359,92 @@ parse_comment(
     ++cs;
 do_com1:
     if(BOOST_JSON_UNLIKELY(! cs))
-    {
-        if(BOOST_JSON_LIKELY(more_))
-            suspend(state::com1);
-        return result::partial;
-    }
+        return maybe_suspend(cs.begin(), state::com1);
     switch(*cs)
     {
     case '/':
         ++cs;
 do_com2:
+        // KRYSTIAN TODO: this is a mess, we have to fix this
         remain = cs.remain();
-        end_tok = remain ? static_cast<const char*>(
-            std::memchr(cs.data(), '\n', remain)) : nullptr;
-        if(BOOST_JSON_UNLIKELY(! end_tok))
+        cs = remain ? static_cast<const char*>(
+            std::memchr(cs.begin(), '\n', remain)) : canary();
```

Comment:
no ternary please

Reply:
Going to address this in another commit, later on